i18n-embed-fl/src/lib.rs

```diff
@@ -801,56 +808,101 @@ fn check_attribute_args(
+) where
+    R: std::borrow::Borrow<FluentResource>,
+{
     pattern.elements.iter().for_each(|element| {
         if let PatternElement::Placeable { expression } = element {
-            args_from_expression(expression, args)
+            args_from_expression(expression, bundle, args)
         }
     });
 }
 
-fn args_from_expression<S: Copy>(expr: &Expression<S>, args: &mut Vec<S>) {
+fn args_from_expression<'m, R>(
+    expr: &Expression<&'m str>,
+    bundle: &'m FluentBundle<R>,
+    args: &mut Vec<&'m str>,
+) where
+    R: std::borrow::Borrow<FluentResource>,
+{
     match expr {
         Expression::Inline(inline_expr) => {
-            args_from_inline_expression(inline_expr, args);
+            args_from_inline_expression(inline_expr, bundle, args);
         }
         Expression::Select { selector, variants } => {
-            args_from_inline_expression(selector, args);
+            args_from_inline_expression(selector, bundle, args);
 
             variants.iter().for_each(|variant| {
-                args_from_pattern(&variant.value, args);
+                args_from_pattern(&variant.value, bundle, args);
             })
         }
     }
 }
 
-fn args_from_inline_expression<S: Copy>(inline_expr: &InlineExpression<S>, args: &mut Vec<S>) {
+fn args_from_inline_expression<'m, R>(
+    inline_expr: &InlineExpression<&'m str>,
+    bundle: &'m FluentBundle<R>,
+    args: &mut Vec<&'m str>,
+) where
+    R: std::borrow::Borrow<FluentResource>,
+{
     match inline_expr {
         InlineExpression::FunctionReference {
             id: _,
             arguments: call_args,
         } => {
-            args_from_call_arguments(call_args, args);
+            args_from_call_arguments(call_args, bundle, args);
         }
         InlineExpression::TermReference {
             id: _,
             attribute: _,
             arguments: Some(call_args),
         } => {
-            args_from_call_arguments(call_args, args);
+            args_from_call_arguments(call_args, bundle, args);
         }
         InlineExpression::VariableReference { id } => args.push(id.name),
-        InlineExpression::Placeable { expression } => args_from_expression(expression, args),
+        InlineExpression::Placeable { expression } => {
+            args_from_expression(expression, bundle, args)
+        }
+        InlineExpression::MessageReference {
+            id,
+            attribute: None,
+        } => {
+            bundle
+                .get_message(&id.name)
+                .and_then(|m| m.value())
+                .map(|p| args_from_pattern(p, bundle, args));
+        }
+        InlineExpression::MessageReference {
+            id,
+            attribute: Some(attribute),
+        } => {
+            bundle
+                .get_message(&id.name)
+                .and_then(|m| m.get_attribute(&attribute.name))
+                .map(|m| m.value())
+                .map(|p| args_from_pattern(p, bundle, args));
+        }
         _ => {}
```

Comment:
Should we go ahead and match this enum exhaustively? The [source enum](https://github.com/projectfluent/fluent-rs/blob/10302d0adc643aa345a432f12eae9da61e290147/fluent-syntax/src/ast/mod.rs#L1010) is not marked `#[non_exhaustive]`, which means that any additions on fluent's part can be considered breaking. Making our match exhaustive will detect any future additions to the AST with breakage here.

Reply:
Thanks @cbs228 that's a great observation! Let's do this in a future PR perhaps? I've created an issue for it https://github.com/kellpossible/cargo-i18n/issues/148